velox/exec/tests/utils/ExpressionBuilder.h

```diff
@@ -0,0 +1,311 @@
+  return {std::make_shared<ConstantExpr>(REAL(), value, std::nullopt)};
+}
+
+inline ExprWrapper literal(const std::string& value) {
+  return {std::make_shared<ConstantExpr>(VARCHAR(), value, std::nullopt)};
+}
+
+// Timestamp literal.
+inline Timestamp parseTimestamp(const std::string& timestamp) {
+  return fromTimestampString(
+             StringView(timestamp), util::TimestampParseMode::kPrestoCast)
+      .thenOrThrow(folly::identity, [&](const Status& status) {
+        VELOX_USER_FAIL("{}", status.message());
+      });
+}
+
+inline ExprWrapper literalTimestamp(const std::string& value) {
+  return {std::make_shared<ConstantExpr>(
+      TIMESTAMP(), variant::timestamp(parseTimestamp(value)), std::nullopt)};
+}
+
+// Unpack a parameter. Either builds a ConstantExpr (literal) based on a scalar
+// value, or passes through an ExprWrapper already constructed.
+template <typename T>
+inline std::shared_ptr<IExpr> unpack(T value) {
+  return literal(value).ptr;
+}
+
+template <>
+inline std::shared_ptr<IExpr> unpack<ExprWrapper>(ExprWrapper expr) {
+  return expr.ptr;
+}
+
+// Unpacks a list of variadic template parameters in a std::vector<ExprPtr>. The
+// elements could be ExprWrapper or C++ literals, which will get converted to
+// ConstantExpr.
+//
+// Base of recursion.
+inline std::vector<ExprPtr> unpackList() {
+  return {};
+}
+
+template <typename TFirst, typename... TArgs>
+inline std::vector<ExprPtr> unpackList(TFirst first, TArgs&&... args) {
+  std::vector<ExprPtr> head = {unpack(first)};
+  auto tail = unpackList(std::forward<TArgs>(args)...);
+  head.insert(head.end(), tail.begin(), tail.end());
+  return head;
+}
+
+// Macro to reduce builerplate when overloading C++ operators. The template
+// magic basically means that the overload is matched if either left or right
+// operands are an ExprWrapper. This is so that both "c"_f + 10 and 10 + "c"_f
+// are supported, for example.
+//
+// If either left or right side are ExprWrapper, we either convert the other
+// side as a constant/literal, or use it as-is if it is already an ExprWrapper.
+#define VELOX_EXPR_BUILDER_OPERATOR(__op, __name)                         \
+  template <typename T1, typename T2>                                     \
+  inline std::enable_if_t<                                                \
+      std::is_same_v<T1, ExprWrapper> || std::is_same_v<T2, ExprWrapper>, \
+      ExprWrapper>                                                        \
+  __op(T1 lhs, T2 rhs) {                                                  \
+    return {std::make_shared<CallExpr>(                                   \
+        __name,                                                           \
+        std::vector<ExprPtr>{unpack(lhs), unpack(rhs)},                   \
+        std::nullopt)};                                                   \
+  }
+
+// Define C++ operator overload for comparisons (filters).
+VELOX_EXPR_BUILDER_OPERATOR(operator==, "eq");
```

Comment:
We have namespace prefix for functions (e.g. `presto.default.eq`) so this need to be configurable.  Maybe a global variable for prefix is enough. But I would guess there will be other things people want to configure in the future. So a global singleton of config object probably is better.

Reply:
+1. Besides the prefix, I believe Spark functions have different names, e.g. 'eq' in Presto is called 'equalto' in Spark.